persist/sqlite/accounts.go

```diff
@@ -212,6 +348,65 @@ func contractFunding(tx *txn, accountID int64) (fund []fundAmount, err error) {
 	return
 }
 
+// updateV2ContractFunding distributes account usage to the contracts that funded
+// the account.
+func updateV2ContractFunding(tx *txn, accountID int64, usage proto4.Usage) error {
```

Comment:
This seems a bit complicated for a solution that results in inaccurate information. It might be simpler and more accurate to always credit the last contract that was used for funding an account.
Or to not associate usage with contracts and instead with accounts.

Reply:
Also the name is not consistent with `updateContractUsage` which seems to be the same thing but for v1?

Reply:
Associating usage with an account is unacceptable UX.

Reply:
The UX I imagine is 
1. click on contract
2. contract shows you contract spending and a list of accounts and amounts that it was used to fund with a number indicating the amount. Most of the time that will be a single account.
3. you click the account to get detailed spending metrics for it and a list of contracts that were used to fund it for you to seamlessly navigate between contracts and accounts

If you spread it out across contracts you end up with some awkward UX depending on whether you also include expired contracts or not. Either you increase the spending of contracts which are expired which isn't great or you potentially end up seemingly spending more money from a contract than it contained to begin with.

Reply:
That UX is significantly more complex than associating it with the contract  in the first place, which is a more accurate representation and what users are expecting.